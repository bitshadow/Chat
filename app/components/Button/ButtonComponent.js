// @flow
import React, { Component } from 'react';
import type { TouchableWithoutFeedback } from 'react-native';
import { Button } from 'react-native-elements';
import colors from '../../utils/colors';
import { scale } from '../../utils/styles';

type ButtonProps = TouchableWithoutFeedback & {
  title: string,
  onPress: () => void,
  type?: 'primary' | 'secondary' | 'tertiary' | 'quaternary',
  block?: boolean,
  // provide a SimpleLineIcons name from:
  // https://oblador.github.io/react-native-vector-icons/
  icon?: string,
  iconType?: string,
  iconSize?: number,
  fontSize?: number,
  large?: boolean,
  disabled?: boolean,
  loading?: boolean,
};

class ButtonComponent extends Component<ButtonProps, *> {
  static defaultProps = {
    title: '',
    type: 'primary',
    block: false,
    iconSize: 18,
    iconType: 'simple-line-icon',
    fontSize: 15,
  };

  getColor = () => {
    switch (this.props.type) {
      case 'secondary':
        return colors.blue;
      case 'tertiary':
        return colors.orange;
      case 'quaternary':
        return colors.lightGreen;
      default:
        return colors.pink;
    }
  };

  render() {
    const {
      title,
      icon,
      iconSize,
      iconType,
      fontSize,
      block,
      buttonStyle,
      containerViewStyle,
      ...rest
    } = this.props;

    return (
      <Button
        title={title.toUpperCase()}
        raised={!this.props.disabled && !this.props.loading}
        buttonStyle={{
          paddingVertical: scale(14),
          paddingHorizontal: scale(40),
          ...buttonStyle,
        }}
        containerViewStyle={
          block
            ? {
                width: '100%',
                marginLeft: 0,
                marginRight: 0,
                ...containerViewStyle,
              }
            : { ...containerViewStyle }
        }
        borderRadius={3}
        backgroundColor={this.getColor().hex()}
        disabledStyle={{
          backgroundColor: this.getColor().desaturate(0.3).lighten(0.3).hex(),
        }}
        fontWeight="bold"
        fontSize={scale(fontSize)}
        rightIcon={{
          name: icon,
          type: iconType,
          size: iconSize,
        }}
        {...rest}
      />
    );
  }
}

export default ButtonComponent;
