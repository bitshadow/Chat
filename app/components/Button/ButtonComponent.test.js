import 'react-native';
import 'isomorphic-fetch';
import React from 'react';
import renderer from 'react-test-renderer';
import ButtonComponent from './ButtonComponent';

test('renders correctly', () => {
  const tree = renderer.create(<ButtonComponent />).toJSON();

  expect(tree).toMatchSnapshot();
});
