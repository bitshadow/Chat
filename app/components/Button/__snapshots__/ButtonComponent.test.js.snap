// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "marginLeft": 15,
        "marginRight": 15,
      },
      Object {
        "shadowColor": "rgba(0,0,0, .4)",
        "shadowOffset": Object {
          "height": 1,
          "width": 1,
        },
        "shadowOpacity": 1,
        "shadowRadius": 1,
      },
      Object {},
      Object {
        "borderRadius": 3,
      },
    ]
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      pointerEvents="box-only"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#9E9E9E",
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 19,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Object {
            "backgroundColor": "#E05790",
          },
          Object {
            "borderRadius": 3,
          },
          Object {
            "padding": 12,
          },
          undefined,
          undefined,
          undefined,
          Object {
            "paddingHorizontal": 40,
            "paddingVertical": 14,
          },
          undefined,
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Array [
              Object {
                "color": "white",
                "fontSize": 20,
              },
              undefined,
              Object {
                "fontSize": 17.5,
              },
              Object {
                "fontSize": 15,
              },
              undefined,
              Object {
                "fontWeight": "bold",
              },
              undefined,
              undefined,
            ],
          ]
        }
      >
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 18,
            },
            Array [
              Object {
                "marginLeft": 10,
              },
              undefined,
            ],
            Object {
              "fontFamily": "simple-line-icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
        type="simple-line-icon"
      >
        
      </Text>
    </View>
  </View>
</View>
`;
